homework/list.js

```diff
@@ -1,79 +1,85 @@
 'use strict';
 
-(function() {
-
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
+    var ListNoteManager = function() {
+        var _listNoteContent = [];
+        var _wrapper = null;
+    };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    ListNoteManager.prototype = {
+        start() {
+            this._wrapper = document.querySelector('#note-list-wrapper');
+            this.fetchList().then((function(data) {
+                this.updateList(data);
+                this.drawList();
+                this.preloadFirstNote();
+            }).bind(this));
+            window.addEventListener('click', (function(event) {
+                this.onNoteOpen(event);
+            }).bind(this));
+        },
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+        onNoteOpen(event) {
+            if (event.target.classList.contains('note-title')) {
```

Comment:
In fact it's a convention to prefix a handler with `on` to the corresponding event. So it should be renamed as `onNoteClicked`.